File: services/api/transaction.js
import { apiRequest } from './_request.js';

export function reserveBook(payload) {
  return apiRequest('/api/transaction/reserve', { method: 'POST', body: payload });
}

export function borrowBook(payload) {
  return apiRequest('/api/transactions/borrow', { method: 'POST', body: payload });
}

export function returnBook(payload) {
  return apiRequest('/api/transactions/return', { method: 'POST', body: payload });
}

export function forceReturn(payload) {
  return apiRequest('/api/transactions/force-return', { method: 'POST', body: payload });
}

export function cancelReservation(transactionId) {
  return apiRequest('/api/transactions/cancel', { method: 'POST', body: { transaction_id: transactionId } });
}

export function getManageTransactions(params = {}) {
  return apiRequest('/api/transactions/manage', { params });
}

export function notifyBorrower(payload) {
  return apiRequest('/api/transactions/notify-borrower', { method: 'POST', body: payload });
}

File: src/api/transaction.py
"""Transaction API blueprint and compatibility exports."""

from flask import Blueprint

from src.api.transactions import (
    borrow_book,
    cancel_reservation,
    force_return,
    get_book_effective_status,
    get_book_history,
    get_manage_transactions,
    notify_borrower,
    reserve_book,
    return_book,
)

transaction_bp = Blueprint('transaction_api', __name__)
transaction_bp.add_url_rule('/api/transactions/reserve', 'reserve_book', reserve_book, methods=['POST'])
transaction_bp.add_url_rule('/api/transaction/reserve', 'reserve_book_singular', reserve_book, methods=['POST'])
transaction_bp.add_url_rule('/api/transactions/borrow', 'borrow_book', borrow_book, methods=['POST'])
transaction_bp.add_url_rule('/api/transactions/return', 'return_book', return_book, methods=['POST'])
transaction_bp.add_url_rule('/api/transactions/force-return', 'force_return', force_return, methods=['POST'])
transaction_bp.add_url_rule('/api/transactions/cancel', 'cancel_reservation', cancel_reservation, methods=['POST'])
transaction_bp.add_url_rule('/api/transactions/manage', 'get_manage_transactions', get_manage_transactions, methods=['GET'])
transaction_bp.add_url_rule('/api/transactions/notify-borrower', 'notify_borrower', notify_borrower, methods=['POST'])
transaction_bp.add_url_rule('/api/books/history', 'get_book_history', get_book_history, methods=['GET'])

__all__ = [
    'borrow_book',
    'cancel_reservation',
    'force_return',
    'get_book_effective_status',
    'get_book_history',
    'get_manage_transactions',
    'notify_borrower',
    'reserve_book',
    'return_book',
    'transaction_bp',
]
